hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java

```diff
@@ -112,6 +112,10 @@ public Response list(
     ContinueToken decodedToken =
         ContinueToken.decodeFromString(continueToken);
 
+    // stand for startAfter
+    if (startAfter == null) {
+      startAfter = marker;
+    }
```

Comment:
After startAfter = marker, could startAfter be null?
We may remove duplicated check for startAfter after this null checker.

Reply:
@timmylicheng Thanks for your review.

the marker can be `null`, so the startAfter can be `null`, although we assigned the marker value to startAfter.

Reply:
So why do we put marker to startAfter when it's null? Maybe it would be great to have comments to explain this. :)

Reply:
@timmylicheng 
OK, I add a null checker, assign marker to startAfter only when marker != null.

PTAL.